File: backend/app/schemas/signoff_requests.py
"""Schemas for the unified sign-off review queue (F-0080)."""

from datetime import datetime
from typing import Literal, Optional
from uuid import UUID

from pydantic import BaseModel


class QueueActorRef(BaseModel):
    id: str
    name: str
    email: str


class SignoffRequestItem(BaseModel):
    type: Literal["run", "protocol"]
    request_id: Optional[UUID] = None
    target_id: UUID
    target_name: str
    role: Optional[str] = None
    project_id: Optional[UUID] = None
    assigned: bool
    requested_by: Optional[QueueActorRef] = None
    created_at: Optional[datetime] = None

File: frontend/src/lib/schemas/signoffRequests.ts
import { z } from 'zod';

export const QueueActorRefSchema = z.object({
    id: z.string(),
    name: z.string(),
    email: z.string(),
}).passthrough();
export type QueueActorRef = z.infer<typeof QueueActorRefSchema>;

export const SignoffRequestItemSchema = z
    .object({
        type: z.enum(['run', 'protocol']),
        request_id: z.string().nullable().optional(),
        target_id: z.string(),
        target_name: z.string(),
        role: z.string().nullable().optional(),
        project_id: z.string().nullable().optional(),
        assigned: z.boolean(),
        requested_by: QueueActorRefSchema.nullable().optional(),
        created_at: z.string().nullable().optional(),
    })
    .passthrough();
export type SignoffRequestItem = z.infer<typeof SignoffRequestItemSchema>;

export const SignoffRequestListSchema = z.array(SignoffRequestItemSchema);
export type SignoffRequestList = z.infer<typeof SignoffRequestListSchema>;
